### ISSUE TITLE
EICAR is not being detected when attached to a PDF

### ISSUE THREAD

**Issue by BisekJM** [Cisco-Talos/clamav#918]
When I scan an EICAR test document (eicar-download.txt), ClamAV detects the virus.  Note: I copied the EICAR string from eicar.com
When I add that same EICAR test document as an attachment to a PDF file (eicar-download.pdf) and then scan that PDF file with ClamAV, the virus is not detected.
Also, when I include that same EICAR test document in a .zip archive (eicar-download.zip) and then scan the archive file with ClamAV the virus is not detected.

I can't figure this out.  Any help that you could give me would be appreciated.  Even if you could scan these files in your ClamAV environment to verify that the EICAR string is detected.
Thanks!

[clamconf.txt](https://github.com/Cisco-Talos/clamav/files/11410662/clamconf.txt)
[eicar-download.txt](https://github.com/Cisco-Talos/clamav/files/11410664/eicar-download.txt)
[eicar-download.zip](https://github.com/Cisco-Talos/clamav/files/11410667/eicar-download.zip)
[eicar-download.pdf](https://github.com/Cisco-Talos/clamav/files/11410671/eicar-download.pdf)

**Comment by micahsnyder** [comment]
Hi @BisekJM,

At a glance I don't see anything unusual in your clamconf.txt.  If you use `clamconf -n` it will minimize to just the non-default settings, which could help make it easier to look for some hint.

How are you doing the scan?  Simple `clamscan` or something else? 

In my own testing with ClamAV 1.1.0 and ClamAV 0.103.8 I had no issue detecting all three files:
```
❯ ~/clams/0.103.8/bin/clamscan -d ~/.cvdupdate/database ~/tmp/eicar-download.*
/home/micah/tmp/eicar-download.pdf: Win.Test.EICAR_HDB-1 FOUND
/home/micah/tmp/eicar-download.txt: Win.Test.EICAR_HDB-1 FOUND
/home/micah/tmp/eicar-download.zip: Win.Test.EICAR_HDB-1 FOUND

----------- SCAN SUMMARY -----------
Known viruses: 8665533
Engine version: 0.103.8
Scanned directories: 0
Scanned files: 3
Infected files: 3
Data scanned: 0.53 MB
Data read: 0.10 MB (ratio 5.23:1)
Time: 20.316 sec (0 m 20 s)
Start Date: 2023:05:05 17:43:45
End Date:   2023:05:05 17:44:06

```

**Comment by BisekJM** [comment]
Hi @micahsnyder ,
Thank you so much for your really quick response.
Our apps, including ClamAV and Clammit, run in containers in OpenShift.  We POST to Clammit, to scan files.
Today I copied several test files, including the ones that I sent to you, to the OpenShift pod where the ClamAV container is running so that I could do a scan via clamscan and the EICAR virus was detected in all 5 files.

Does clamscan do the same scan that would occur via Clammit?  I tried a clamdscan and got different (somewhat confusing) results.  Maybe I didn't use it correctly.  Results are below.

sh-4.4$ clamscan -i -v 
Scanning /tmp/eicar/eicar-adobe-acrobat-attachment.pdf
/tmp/eicar/eicar-adobe-acrobat-attachment.pdf: Win.Test.EICAR_HDB-1 FOUND
Scanning /tmp/eicar/eicar-download.pdf
/tmp/eicar/eicar-download.pdf: Win.Test.EICAR_HDB-1 FOUND
Scanning /tmp/eicar/eicar-download.txt
/tmp/eicar/eicar-download.txt: Win.Test.EICAR_HDB-1 FOUND
Scanning /tmp/eicar/eicar-download.zip
/tmp/eicar/eicar-download.zip: Win.Test.EICAR_HDB-1 FOUND
Scanning /tmp/eicar/pdf-doc-vba-eicar-dropper.pdf
/tmp/eicar/pdf-doc-vba-eicar-dropper.pdf: Doc.Dropper.Agent-1540415 FOUND

----------- SCAN SUMMARY -----------
Known viruses: 8665533
Engine version: 0.103.8
Scanned directories: 1
Scanned files: 5
Infected files: 5
Data scanned: 0.59 MB
Data read: 0.11 MB (ratio 5.21:1)
Time: 26.521 sec (0 m 26 s)
Start Date: 2023:05:08 18:58:14
End Date:   2023:05:08 18:58:40
sh-4.4$

sh-4.4$ clamdscan -v

----------- SCAN SUMMARY -----------
Infected files: 0
Total errors: 1
Time: 0.000 sec (0 m 0 s)
Start Date: 2023:05:08 19:40:07
End Date:   2023:05:08 19:40:07
sh-4.4$  clamdscan -V
ClamAV 0.103.8
sh-4.4$

[clamconf-n.txt](https://github.com/Cisco-Talos/clamav/files/11424423/clamconf-n.txt)
